pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,625 @@
+// api.pb.go could be generate by hack/update-generate-runtime.sh
+syntax = 'proto2';
+
+package runtime;
+
+// Runtime service defines the public APIs for remote container runtimes
+service RuntimeService {
+    // Version returns the runtime name, runtime version and runtime API version
+    rpc Version(VersionRequest) returns (VersionResponse) {}
+
+    // CreatePodSandbox creates a pod-level sandbox. PodSandbox is interpreted 
+    // differently in different runtimes.  For docker, this may mean 
+    // creating the infra container. For hyper, this can create the 
+    // virtual machine.
+    rpc CreatePodSandbox(CreatePodSandboxRequest) returns (CreatePodSandboxResponse) {}
+    // DeletePodSandbox deletes the sandbox.
+    rpc DeletePodSandbox(DeletePodSandboxRequest) returns (DeletePodSandboxResponse) {}
+    // PodSandboxStatus returns the Status of the PodSandbox.
+    rpc PodSandboxStatus(PodSandboxStatusRequest) returns (PodSandboxStatusResponse) {}
+    // ListPodSandbox returns a list of SandBox.
+    rpc ListPodSandbox(ListPodSandboxRequest) returns (ListPodSandboxResponse) {}
+
+    // CreateContainer creates a new container in specified PodSandbox
+    rpc CreateContainer(CreateContainerRequest) returns (CreateContainerResponse) {}
+    // StartContainer starts the container.
+    rpc StartContainer(StartContainerRequest) returns (StartContainerResponse) {}
+    // StopContainer stops a running container with a grace period (i.e., timeout).
+    rpc StopContainer(StopContainerRequest) returns (StopContainerResponse) {}
+    // RemoveContainer removes the container.
+    rpc RemoveContainer(RemoveContainerRequest) returns (RemoveContainerResponse) {}
+    // ListContainers lists all containers by filters.
+    rpc ListContainers(ListContainersRequest) returns (ListContainersResponse) {}
+    // ContainerStatus returns the container status.
+    rpc ContainerStatus(ContainerStatusRequest) returns (ContainerStatusResponse) {}
+
+    // Exec execute a command in the container.
+    rpc Exec(stream ExecRequest) returns (stream ExecResponse) {}
+}
+
+// Image service defines the public APIs for managing images
+service ImageService {
+    // ListImages lists existing images.
+    rpc ListImages(ListImagesRequest) returns (ListImagesResponse) {}
+    // ImageStatus returns the status of the image.
+    rpc ImageStatus(ImageStatusRequest) returns (ImageStatusResponse) {}
+    // PullImage pulls a image with authentication config.
+    rpc PullImage(PullImageRequest) returns (PullImageResponse) {}
+    // RemoveImage removes the image.
+    rpc RemoveImage(RemoveImageRequest) returns (RemoveImageResponse) {}
+}
+
+message VersionRequest {
+    // The version of the gRPC API
+    optional string version = 1;
+}
+
+message VersionResponse {
+    // The version of the gRPC API
+    optional string version = 1;
```

Comment:
Already changed this to '*The version of the kubelet runtime API*.' in
latest commit.


2016-07-07 5:56 GMT+08:00 Yu-Ju Hong <notifications@github.com>:

> In pkg/kubelet/api/v1alpha1/runtime/api.proto
> <https://github.com/kubernetes/kubernetes/pull/17048#discussion_r69818894>
> :
>
> > +    // ImageStatus returns the status of the image.
> > +    rpc ImageStatus(ImageStatusRequest) returns (ImageStatusResponse) {}
> > +    // PullImage pulls a image with authentication config.
> > +    rpc PullImage(PullImageRequest) returns (PullImageResponse) {}
> > +    // RemoveImage removes the image.
> > +    rpc RemoveImage(RemoveImageRequest) returns (RemoveImageResponse) {}
> > +}
> > +
> > +message VersionRequest {
> > +    // The version of the gRPC API
> > +    optional string version = 1;
> > +}
> > +
> > +message VersionResponse {
> > +    // The version of the gRPC API
> > +    optional string version = 1;
>
> @feiskyer <https://github.com/feiskyer> do you mind adding more comments
> to clarify this? Thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17048/files/0978833eb5a548fc32199489e9c62686a1aae035#r69818894>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AApTHcDcr8ylNVIkI406SuiAVj23M-yUks5qTCRwgaJpZM4GfQ_B>
> .
>